packages/go_router/lib/src/parser.dart

```diff
@@ -67,8 +67,26 @@ class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
       // This is a result of browser backward/forward button or state
       // restoration. In this case, the route match list is already stored in
       // the state.
-      final RouteMatchList matchList =
+      final RouteMatchList baseMatchList =
           _routeMatchListCodec.decode(state as Map<Object?, Object?>);
+      final Uri? newUri = baseMatchList.matches
```

Comment:
If there is a imperative route match, should we only redirect that imperative route match instead of redirect the entire baseMatchList?